### PR TITLE
Migrations for Topics, Modules, Questions, Feedback 

### LINKED ISSUE #63: Databases: Create Migrations for Topics, Modules, Questions, Feedback and Users
Create the Knex Migrations in order to create the DB schema for the following tables:

Topics, Modules, Questions, Feedback and Users - Topic Yellow 

Screenshot 2020-11-22 at 15 21 34

Please consider the DB Diagram - https://dbdiagram.io/d/5fb801913a78976d7b7cc7d8

### PR DESCRIPTION
**Pull request by yukafjaponesa**
# Description

Created Knex migrations files which consist of DB tables for all Topics, Modules, Questions, Feedback and Users - with team Yellow includes (@1620prabn , @Elia-Git , @yukafjaponesa )

Fixes #63 

# How to test?

in text editor.

# Checklist

- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have made corresponding changes to the documentation
- [ x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by dpfernandes — CHANGES_REQUESTED** [review]
deleted_at should be null.

This field is used to validate if the record inserted was deleted or not. The logic behind is if deleted_at has a date it's because that same record was deleted on that same date.

Check this article for more info https://medium.com/galvanize/soft-deletion-is-actually-pretty-hard-cb434e24825c

**Comment by dpfernandes** [comment]
If this PR has 4 migrations files for 4 different tables why do you call it **Db migration modules** ?

**Comment by dpfernandes** [comment]
Also the description of the PR only mentions Modules when you have 3 tables more. Why is that?

**Comment by 1620prabn** [comment]
> If this PR has 4 migrations files for 4 different tables why do you call it **Db migration modules** ?

titile is changed

File: src/server/migrations/20201125222522_create_questions_table.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+exports.up = function(knex) {
+  return knex.schema.createTable('questions', (table) => {
+    table.increments();
+    table
+      .integer('topic')
+      .references('id')
+      .inTable('topics');
+    table
+      .integer('asked_by')
+      .references('id')
```
Comment (tpetrina) [inline src/server/migrations/20201125222522_create_questions_table.js:10]: Missing `.unsigned()` before `references`.

File: src/server/migrations/20201125222522_create_questions_table.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      .inTable('users');
+    table
+      .integer('event_id')
+      .references('id')
```
Comment (tpetrina) [inline src/server/migrations/20201125222522_create_questions_table.js:14]: Missing `.unsigned()` before `references`.

File: src/server/migrations/20201125223036_create_feedback_table.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+exports.up = function(knex) {
+  return knex.schema.createTable('feedback', (table) => {
+    table.increments();
+    table
+      .integer('student_id')
+      .references('id')
```
Comment (tpetrina) [inline src/server/migrations/20201125223036_create_feedback_table.js:6]: Missing `.unsigned()` before `references`.

File: src/server/migrations/20201125223036_create_feedback_table.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      .inTable('users');
+    table
+      .integer('answered_by')
+      .references('id')
```
Comment (tpetrina) [inline src/server/migrations/20201125223036_create_feedback_table.js:10]: Missing `.unsigned()` before `references`.

File: src/server/migrations/20201125223036_create_feedback_table.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      .inTable('users');
+    table.string('open_feedback').notNullable();
+    table.datetime('endDate').notNullable();
+    table
+      .integer('question_id')
+      .references('id')
```
Comment (tpetrina) [inline src/server/migrations/20201125223036_create_feedback_table.js:16]: Missing `.unsigned()` before `references`.

**Review by tpetrina — CHANGES_REQUESTED** [review]
I am a bit disappointed in this not working at all. You must try running the code or asking for help if it doesn't work.

**Comment by tpetrina** [comment]
But to not sound too negative, this is a 90% great job. You were almost at the finish line! 👍

File: src/server/migrations/20201129173036_create_feedback_table.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
+exports.up = function (knex) {
```
Comment (Elia-Git) [inline src/server/migrations/20201129173036_create_feedback_table.js:1]: This table should be deleted also in the remote. You only deleted locally but not in the remote.

File: src/server/migrations/20201129223036_create_feedback_table.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    table.string('description').notNullable();
+    table.datetime('created_at').defaultTo(knex.fn.now()).notNullable();
+    table.datetime('updated_at').defaultTo(knex.fn.now()).notNullable();
+    table.datetime('deleted_at');
+  });
+};
+
+exports.down = function (knex) {
+  return knex.schema.dropTable('feedback');
+};
```
Comment (Elia-Git) [inline src/server/migrations/20201129223036_create_feedback_table.js:22]: This table also needs to be deleted !

File: src/server/migrations/20201129173036_create_feedback_table.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    table.string('description').notNullable();
+    table.datetime('created_at').defaultTo(knex.fn.now()).notNullable();
+    table.datetime('updated_at').defaultTo(knex.fn.now()).notNullable();
+    table.datetime('deleted_at');
+  });
+};
+
+exports.down = function (knex) {
+  return knex.schema.dropTable('feedback');
+};
```
Comment (Elia-Git) [inline src/server/migrations/20201129173036_create_feedback_table.js:22]: This table should be deleted also in the remote. You only deleted locally but not in the remote. 
As Questions table is created this time 20201129222522

**Pull request closed without merging**